### ISSUE TITLE
Sprite atifacts and offset in TextureAtlasSprites

### ISSUE THREAD

**Issue by NiklasEi** [bevyengine/bevy#3593]
## Bevy version

0.6

## Operating system & version

Ubuntu 18.04

## What you did

Animate a sprite using a texture atlas

## What you expected to happen
![fine](https://user-images.githubusercontent.com/12236672/148654781-5340e3a9-0ed8-4557-9c35-1a9054ee177f.png)

## What actually happened
https://user-images.githubusercontent.com/12236672/148654830-7cd5f5a8-6f14-4c84-8f94-86f6b8eaafa4.mp4

## Additional information
I tracked this change down to c2da7800e3671ad92e775529070a814d0bc2f5f8

**Comment by Davier** [comment]
The png is 768x99 pixels, loading the atlas with these parameters works as expected:
`#[asset(texture_atlas(tile_size_x = 96., tile_size_y = 96., columns = 8, rows = 1))]`.
I suspect that before c2da780 the sprites were rescaled to 100*96.

**Comment by NiklasEi** [comment]
Yeah, I just found that bug too and fixed it. That doesn't fix the render artifacts though.

**Comment by Davier** [comment]
> Yeah, I just found that bug too and fixed it. That doesn't fix the render artefacts though.

That's strange, [this commit](https://github.com/NiklasEi/bevy_asset_loader/tree/e699d0b75051e649d484522c4e561ff8e9c910f8) did fix it for me.

**Comment by Davier** [comment]
Actually, I don't have any artefacts on the previous commit (v0.8.0) either, it only had the size issue. I can't reproduce. Could you detail your OS and GPU?

**Comment by NiklasEi** [comment]
I can also reproduce the artefacts with the `sprite` example

on 32f7997c566f4dcf8dc57b64f20193234256deb1
![fine](https://user-images.githubusercontent.com/12236672/148658260-833eba21-8bfb-4f28-9dbc-cd873a9e2b97.png)

on c2da7800e3671ad92e775529070a814d0bc2f5f8
![issue](https://user-images.githubusercontent.com/12236672/148658276-b5bb1c87-7e87-425a-8b1c-d065f387fdf5.png)

**Comment by NiklasEi** [comment]
> Actually, I don't have any artefacts on the previous commit (v0.8.0) either, it only had the size issue. I can't reproduce. Could you detail your OS and GPU?

I added some `vulkaninfo` output to the PR description.

**Comment by rparrett** [comment]
Is this related to #3423?

Not seeing this on windows/nvidia or mac/m1

**Comment by Davier** [comment]
I think the only rendering change for sprites in this commit is enabling MSAA. Could you try to disable it?
```rust
app.insert_resource(Msaa { samples: 1 })
```

**Comment by NiklasEi** [comment]
> I think the only rendering change for sprites in this commit is enabling MSAA. Could you try to disable it?
> 
> ```rust
> app.insert_resource(Msaa { samples: 1 })
> ```

That fixes it for me

**Comment by lemunozm** [comment]
Hi! Same issue here. 
- Ubuntu 20.04.02. 
- Adapter: Intel(R) UHD Graphics 620 (WHL GT2).

But I also notice that the background darkens with msaa enabled.

msaa 1:
![msaa-1](https://user-images.githubusercontent.com/15687891/154453382-3841289d-9479-4069-ae04-ab6204dca4f3.png)

msaa 4: 
![msaa-4](https://user-images.githubusercontent.com/15687891/154453428-30f712e8-2b2d-457c-9de2-1fd30266241b.png)

msaa 2 and 8 values make the application crash at starting

**Comment by johanhelsing** [comment]
I'm seeing the same thing on both wayland and x11, but not wasm (also linux, same machine) and not on windows.

![image](https://user-images.githubusercontent.com/2620557/172384553-e3b9cc69-319b-455f-bc40-0874c240c7ac.png)

Areas of the screen where there is no blending going on are too dark. Looks right when I set MSAA to 1.

I'm also on Intel UHD.

Let me know if there is anything I can add/do to help.

**Comment by devil-ira** [comment]
The darkened colors suggest this issue may have been caused by Bevy picking an incorrect surface format, which should be fixed on `main`.
If anyone affected could test it again, that would be great.

**Comment by JamesGallagherPoole** [comment]
This is happening for me too. I am getting darkened colours and artefacts before and after updating to 0.11.

I'm on a linux laptop with Intel UHD. Setting Msaa to off fixes the problem.

Example of with Msaa:
![image](https://github.com/bevyengine/bevy/assets/12208147/a8cd7440-342d-40f8-be19-ded473489115)

Example of without Msaa:
![image](https://github.com/bevyengine/bevy/assets/12208147/bd3f9fa2-a031-4aa9-ac5a-8b9b8bb8a948)

**Comment by billymosis** [comment]
same here. I'm on older hardware
before disabled:
![Bevy App_003](https://github.com/bevyengine/bevy/assets/57342180/ad278dbf-7117-4adc-80de-6e495073d4c8)
```rust
fn main() {
    App::new()
        .insert_resource(Msaa::Off)
}
```
after:
![Bevy App_002](https://github.com/bevyengine/bevy/assets/57342180/406ee04e-7c99-4755-8aaa-5c3c141c5119)
spec:
![Selection_545](https://github.com/bevyengine/bevy/assets/57342180/57bb2006-4094-42da-8d0b-99e3603e9102)

**Comment by JamesGallagherPoole** [comment]
It is for me anyway!

**Comment by mockersf** [comment]
then it looks like https://github.com/gfx-rs/wgpu/issues/725 came back... @cwfitzgerald are you aware if that's the case?

**Comment by NiklasEi** [comment]
I can no longer reproduce this, but my hardware has changed. Can someone with hardware that had the issue in previous versions check 0.12 please?

**Comment by JamesGallagherPoole** [comment]
Confirmed still happening for me on 0.12 with the same Intel Integrated GPU, Pop OS laptop

**Comment by cwfitzgerald** [comment]
For those who are experiencing this on vulkan, can you make sure your graphics drivers are up to date? We use this ppa https://github.com/gfx-rs/wgpu/blob/trunk/.github/workflows/ci.yml#L402 to get the latest stable release of Mesa.

If this is still a problem on an updated Mesa, this is a bug to file against Mesa itself.

**Comment by fatho** [comment]
@cwfitzgerald The issue indeed persists with the latest Mesa (23.1.9 at the time of writing).

I am not sure it is a problem with the driver, however. To have something to compare against, I adapted [`vulkano`](https://github.com/vulkano-rs/vulkano)'s triangle example to use multisampling (https://gist.github.com/fatho/e7e7297ee510980bdb550928e64882aa#file-triangle-rs-L365, based on the latest master commit at the time of writing, https://github.com/vulkano-rs/vulkano/blob/795f02217a569254e53a3970d72ec52cdff39bcb/examples/triangle/main.rs).

As the clear value, I used the same `0.5` gray that I used in my bevy reproducer:

```rust
                            clear_values: vec![Some([0.5, 0.5, 0.5, 1.0].into()), None],
```

And it indeed aligns with the color in the Bevy reproducer without MSAA, but when MSAA is turned on in Bevy, you can see the discrepancy (left is bevy with MSAA, right is Vulkano with MSAA, both with 4 samples):

![image](https://github.com/bevyengine/bevy/assets/3857449/46a69cad-7879-4dc8-839f-f0c8fc0e6268)

I wouldn't call myself a Vulkan expert though, and this was just a quick attempt to reproduce the issue, so I wouldn't rule out any mistakes in my adaption of that triangle example either.
